lib/wharel.rb

```diff
@@ -3,10 +3,12 @@
 
 module Wharel
   module QueryMethods
-    %w[where order having pluck group].each do |method_name|
+    %w[select where order having pluck pick group].each do |method_name|
+      next unless ::ActiveRecord::Base.singleton_class.method_defined?(method_name)
+
       module_eval <<-EOM, __FILE__, __LINE__ + 1
       def #{method_name}(*, &block)
-        block_given? ? super(VirtualRow.build_query(self, &block)) : super
+        block_given? ? super(VirtualRow.build_query(self, &block), &nil) : super
```

Comment:
Wha... `&nil`, what is this black magic? :open_mouth:

Reply:
Oh I see, it explicitly passes no block to `super`, neat! I learned something new.

Reply:
I'm wondering if Wharel should handle the case where `select` is called with a non-zero array of args *and* a block. In that case we could go to `super`, since this appears to be a use-case for `select` which the current code is blocking...